### ISSUE TITLE
How to get the current audio progress over time?

### ISSUE THREAD

**Issue by 5haw4** [spotify/web-playback-sdk#86]
TLDR:
In the Web playback SDK I'm trying to achieve the same functionality as if I were to use the DOM event "timeupdate" and "audio.currentTime" in order to update over time my progress text and show the current progress of the audio.

|
|
|
|

I can't find a simple way of getting the current progress of the audio that's being played in order to update my UI accordingly, and for some reason the web SDK doesn't generate / create an accessible audio tag in the HTML, so I can't just use the DOM listener "timeupdate" and get its "currentTime" in order to get the current progress of the audio every time it's changed...

I tried to use intervals of 1000 milliseconds (also tried less) and in that interval I get the SDK's current state with "player.getCurrentState()" but for some reason sometimes when you stop playing or when you disconnect the instance (usually when you open up Spotify's app on the phone), the SDK still keeps on increasing the "state.position" even though the music is no longer being played on the WEB nor anywhere else, and it's increasing the "state.position" and never stops even when its value is greater than the "state.duration" value...

I'd love to get some help with that bug / unexpected behavior, thanks in advance :)

**Comment by tobika** [comment]
My approach is to also use an interval and correct the time difference with the spotify state that arrives every 30sec via the on_state_change, if the user pauses the playback you will also immediately receive an state change event for this

If you call getCurrentState() all the time you could even get problems with race conditions between the http calls which might be the explanation for the wrong state you sometimes have

**Comment by 5haw4** [comment]
> 
> 
> My approach is to also use an interval and correct the time difference with the spotify state that arrives every 30sec via the on_state_change, if the user pauses the playback you will also immediately receive an state change event for this
> 
> If you call getCurrentState() all the time you could even get problems with race conditions between the http calls which might be the explanation for the wrong state you sometimes have

Can u share your solution? (the few lines of code that responsible on tracking the progress)
